src/renderer/components/search/index.vue

```diff
@@ -0,0 +1,445 @@
+        return this.searchMatches.index
+      } else {
+        return -1
+      }
+    },
+    highlightCount () {
+      if (this.searchMatches) {
+        return this.searchMatches.matches.length
+      } else {
+        return 0
+      }
+    }
+  },
+
+  created () {
+    bus.$on('find', this.listenFind)
+    bus.$on('replace', this.listenReplace)
+    bus.$on('findNext', this.listenFindNext)
+    bus.$on('findPrev', this.listenFindPrev)
+    document.addEventListener('click', this.docClick)
+    document.addEventListener('keyup', this.docKeyup)
+  },
+
+  beforeDestroy () {
+    bus.$off('find', this.listenFind)
+    bus.$off('replace', this.listenReplace)
+    bus.$off('findNext', this.listenFindNext)
+    bus.$off('findPrev', this.listenFindPrev)
+    document.removeEventListener('click', this.docClick)
+    document.removeEventListener('keyup', this.docKeyup)
+  },
+
+  methods: {
+    toggleCtrl (ctrl) {
+      this[ctrl] = !this[ctrl]
+      this.search()
+    },
+
+    listenFind () {
+      this.showSearch = true
+      this.type = 'search'
+      this.$nextTick(() => {
+        this.$refs.search.focus()
+      })
+    },
+
+    listenReplace () {
+      this.showSearch = true
+      this.type = 'replace'
+    },
+
+    listenFindNext () {
+      this.find('next')
+    },
+
+    listenFindPrev () {
+      this.find('prev')
+    },
+
+    docKeyup (event) {
+      if (event.key === 'Escape') {
+        this.emptySearch(true)
+      }
+    },
+
+    docClick () {
+      if (!this.showSearch) return
+      this.emptySearch()
+    },
+
+    emptySearch (selectHighlight = false) {
+      this.showSearch = false
+      const searchValue = this.searchValue = ''
+      this.replaceValue = ''
+      bus.$emit('searchValue', searchValue, { selectHighlight })
+    },
+
+    toggleSearchType () {
+      this.type = this.type === 'search' ? 'replace' : 'search'
+    },
+
+    /**
+     * Find the previous or next search result.
+     * action: prev or next
+     */
+    find (action) {
+      bus.$emit('find', action)
+    },
+
+    search (event) {
+      if (event && event.key === 'Escape') {
+        return
+      }
+
+      if (event && event.key === 'Enter') {
+        return this.find('next')
+      }
+
+      const { searchValue, isCaseSensitive, isWholeWord, isRegexp } = this
+      if (isRegexp) {
+        // Handle invalid regexp.
+        try {
+          new RegExp(searchValue)
+          this.searchErrorMsg = ''
+        } catch (err) {
+          this.searchErrorMsg = `Invalid regular expression: /${searchValue}/.`
+          return
+        }
+        // Handle match empty string, no need to search.
+        try {
+          const SEARCH_REG = new RegExp(searchValue)
+          if (searchValue && SEARCH_REG.test('')) {
+            throw new Error()
+          }
+          this.searchErrorMsg = ''
+        } catch (err) {
+          this.searchErrorMsg = `The regexp: /${searchValue}/ match too more.`
+          return
+        }
```

Comment:
@fxha FYI, I add error notification here.

Reply:
I prefer `too much` instead `too more`.

Reply:
Have changed to `RegExp: /${searchValue}/ match empty string.`.